File: dart_programs/practise_programs/leaders.dart

void main() {
  List<int> arr = [10, 22, 12, 3, 0, 6];
  List<int> leaders = [];
  int max = arr[arr.length - 1];
  leaders.add(max);
  for (int i = arr.length - 2; i >= 0; i--) {
    if (arr[i] > max) {
      leaders.add(arr[i]);
      max = arr[i];
    }
  }
  print(leaders.reversed);
}

File: striver_series/src/Practise/Leaders.java
package Practise;

import java.util.ArrayList;

public class Leaders {
    public static void main(String[] args) {
        int arr[]=  {10, 22, 12, 3, 0, 6};
        int n = arr.length;
        ArrayList<Integer> result = new ArrayList<>();
        int max = arr[n-1];
        result.add(max);
        for(int i = n-2;i>=0;i--){
            if(arr[i]>max){
                max = arr[i];
                result.add(max);
            }
        }
        System.out.println(result);
    }
}
